PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:6 + add:0, change matches:
-    _inherit = 'product.supplierinfo'
-    _inherit = 'product.product'
-    _inherit = 'purchase.order'
-    _inherit = 'purchase.order.line'
-    _inherit = 'analytic.mixin'
-    _inherit = 'res.config.settings'

Total Changes: 17

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/purchase_requisition/models/product.py
+++ b/addons/purchase_requisition/models/product.py
@@ -4,15 +4,15 @@
 from odoo import fields, models
 
 
-class SupplierInfo(models.Model):
-    _inherit = 'product.supplierinfo'
+class ProductSupplierinfo(models.Model):
+    _inherit = ['product.supplierinfo']
 
     purchase_requisition_id = fields.Many2one('purchase.requisition', related='purchase_requisition_line_id.requisition_id', string='Agreement')
     purchase_requisition_line_id = fields.Many2one('purchase.requisition.line')
 
 
 class ProductProduct(models.Model):
-    _inherit = 'product.product'
+    _inherit = ['product.product']
 
     def _prepare_sellers(self, params=False):
         sellers = super(ProductProduct, self)._prepare_sellers(params=params)

--- a/addons/purchase_requisition/models/purchase.py
+++ b/addons/purchase_requisition/models/purchase.py
@@ -8,7 +8,6 @@ from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT, get_lang
 
 
 class PurchaseOrderGroup(models.Model):
-    _name = 'purchase.order.group'
     _description = "Technical model to group PO for call to tenders"
 
     order_ids = fields.One2many('purchase.order', 'purchase_group_id')
@@ -21,7 +20,7 @@ class PurchaseOrderGroup(models.Model):
 
 
 class PurchaseOrder(models.Model):
-    _inherit = 'purchase.order'
+    _inherit = ['purchase.order']
 
     requisition_id = fields.Many2one('purchase.requisition', string='Agreement', copy=False)
     requisition_type = fields.Selection(related='requisition_id.requisition_type')
@@ -260,7 +259,7 @@ class PurchaseOrder(models.Model):
 
 
 class PurchaseOrderLine(models.Model):
-    _inherit = 'purchase.order.line'
+    _inherit = ['purchase.order.line']
 
     price_total_cc = fields.Monetary(compute='_compute_price_total_cc', string="Company Subtotal", currency_field="company_currency_id", store=True)
     company_currency_id = fields.Many2one(related="company_id.currency_id", string="Company Currency")

--- a/addons/purchase_requisition/models/purchase_requisition.py
+++ b/addons/purchase_requisition/models/purchase_requisition.py
@@ -6,7 +6,6 @@ from collections import defaultdict
 
 
 class PurchaseRequisition(models.Model):
-    _name = "purchase.requisition"
     _description = "Purchase Requisition"
     _inherit = ['mail.thread', 'mail.activity.mixin']
     _order = "id desc"
@@ -161,8 +160,7 @@ class PurchaseRequisition(models.Model):
 
 
 class PurchaseRequisitionLine(models.Model):
-    _name = "purchase.requisition.line"
-    _inherit = 'analytic.mixin'
+    _inherit = ['analytic.mixin']
     _description = "Purchase Requisition Line"
     _rec_name = 'product_id'
 

--- a/addons/purchase_requisition/models/res_config_settings.py
+++ b/addons/purchase_requisition/models/res_config_settings.py
@@ -4,6 +4,6 @@ from odoo import fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = 'res.config.settings'
+    _inherit = ['res.config.settings']
 
     group_purchase_alternatives = fields.Boolean("Purchase Alternatives", implied_group='purchase_requisition.group_purchase_alternatives')
